File: dev/test_data/typescript/test_main/aas_core_meta.v3/input/snippets/Verification/lang_strings_have_unique_languages.ts
/**
 * Check that `langStrings` are specified each for a unique
 * language.
 *
 * @param langStrings - to be verified
 * @returns `true` if the check passes
 */
export function langStringsHaveUniqueLanguages(
  langStrings: Iterable<AasTypes.IAbstractLangString>
): boolean {
  const languageSet = new Set<string>();

  for (const langString of langStrings) {
    if (languageSet.has(langString.language)) {
      return false;
    }

    languageSet.add(langString.language);
  }

  return true;
}

File: dev/test_data/test_main/golang/aas_core_meta.v3/input/snippets/Verification/lang_strings_have_unique_languages.go
// Check that `langStrings` are specified each for a unique language.
func LangStringsHaveUniqueLanguages[L aastypes.IAbstractLangString](
	langStrings []L) bool {
	languageSet := make(map[string]struct{})

	for _, langString := range langStrings {
		language := langString.Language()
		_, has := languageSet[language]
		if has {
			return false
		}

		languageSet[language] = struct{}{}
	}

	return true
}
